document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/mapping/smallfull/mappers/ApiToPeriodMapperDecorator.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.mappers;
+
+import uk.gov.companieshouse.accountsdates.AccountsDatesHelper;
+import uk.gov.companieshouse.accountsdates.impl.AccountsDatesHelperImpl;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.company.account.LastAccountsApi;
+import uk.gov.companieshouse.api.model.company.account.NextAccountsApi;
+import uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.model.ixbrl.period.Period;
+
+import java.time.LocalDate;
+
+
+public abstract class ApiToPeriodMapperDecorator implements ApiToPeriodMapper {
+
+    private final ApiToPeriodMapper apiToPeriodMapper;
+
+    private AccountsDatesHelper accountsDatesHelper = new AccountsDatesHelperImpl();
+
+    public ApiToPeriodMapperDecorator(ApiToPeriodMapper apiToPeriodMapper) {
+        this.apiToPeriodMapper = apiToPeriodMapper;
+    }
+
+    @Override
+    public Period apiToPeriod(CompanyProfileApi companyProfile) {
+
+        Period period = apiToPeriodMapper.apiToPeriod(companyProfile);
+
+        boolean isSameYear = isSameYearFiler(companyProfile);
+
+        NextAccountsApi nextAccounts = companyProfile.getAccounts().getNextAccounts();
+        period.setCurrentPeriodStartOnFormatted(convertToDisplayDate(nextAccounts.getPeriodStartOn()));
+        period.setCurrentPeriodEndOnFormatted(convertToDisplayDate(nextAccounts.getPeriodEndOn()));
+        period.setCurrentPeriodBSDate(convertDateToBSDate(nextAccounts.getPeriodStartOn(),
+                nextAccounts.getPeriodEndOn(), isSameYear));
+
+
+        if (isMultipleYearFiler(companyProfile)) {
+            LastAccountsApi lastAccounts = companyProfile.getAccounts().getLastAccounts();
+            period.setPreviousPeriodStartOnFormatted(convertToDisplayDate(lastAccounts.getPeriodStartOn()));
+            period.setPreviousPeriodEndOnFormatted(convertToDisplayDate(lastAccounts.getPeriodEndOn()));
+            period.setPreviousPeriodBSDate(convertDateToBSDate(lastAccounts.getPeriodStartOn(),
+                    lastAccounts.getPeriodEndOn(), isSameYear));
+        }
+
+        return period;
+    }
+
+    private boolean isMultipleYearFiler(CompanyProfileApi companyProfile) {
+        LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+        return lastAccountsApi != null && lastAccountsApi.getPeriodEndOn() != null;
+    }
+
+    private boolean isSameYearFiler(CompanyProfileApi companyProfile) {
+        if (isMultipleYearFiler(companyProfile)) {
+            LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+            NextAccountsApi nextAccountsApi = companyProfile.getAccounts().getNextAccounts();
+            return accountsDatesHelper.isSameYear(lastAccountsApi.getPeriodEndOn(), nextAccountsApi.getPeriodEndOn());
+        }
+        return false;
+    }
+
+    private String convertToDisplayDate(LocalDate date) {
```

Comment:
We could add a comment here to clarify the fomat `YYYY MM dd` .

Reply:
A comment already exists on the dateHelper method it calls